### PR TITLE
New types, updates, cleanup and refactor

### PR DESCRIPTION
**Pull request by AlexxNica**

### CODE REVIEW

File: components/roadmap-grid/RoadmapGrid.tsx [hunk 1]
```diff
@@ -16,37 +16,51 @@ import { GroupWrapper } from './group-wrapper';
 import Header from './header';
 import { Headerline } from './headerline';
 
-export function RoadmapDetailed({ issueData, viewMode }: { issueData: IssueData; viewMode: string }) {
-  const showGroupRowTitle = viewMode === 'detail';
-  console.log('viewMode:', viewMode);
+// const groupData = (data) => '';
```
Comment (SgtPooki) [inline components/roadmap-grid/RoadmapGrid.tsx:19]: remove

File: pages/roadmap-test/[...slug].tsx [hunk 2]
```diff
@@ -1,49 +0,0 @@
-import type { InferGetServerSidePropsType } from 'next';
```
Comment (SgtPooki) [inline pages/roadmap-test/[...slug].tsx:1]: we should leave these. I was working on input to allow devs to encode json necessary to make debugging and development faster.

File: pages/roadmap-test/index.tsx [hunk 3]
```diff
@@ -1,92 +0,0 @@
-import type { NextPage } from 'next';
```
Comment (SgtPooki) [inline pages/roadmap-test/index.tsx:1]: we should leave these. I was working on input to allow devs to encode json necessary to make debugging and development faster.

File: pages/roadmap/[...slug].tsx [hunk 4]
```diff
@@ -33,19 +32,16 @@ export async function getServerSideProps(context) {
 export default function RoadmapPage(props: InferGetServerSidePropsType<typeof getServerSideProps>) {
   console.log('inside /roadmap/[...slug].tsx');
   const { issueData, error, isLocal, view, mode } = props;
+  // console.log('issueData:', issueData);
```
Comment (SgtPooki) [inline pages/roadmap/[...slug].tsx:35]: remove

File: utils/general.ts [hunk 5]
```diff
@@ -31,13 +24,27 @@ export const addOffset = (dates: Date[], { offsetStart, offsetEnd }: { offsetSta
   return datesWithOffset;
 };
 
-export const paramsFromUrl = (url) => {
+//#region -> URL-RELATED
```
Comment (SgtPooki) [inline utils/general.ts:27]: whats this for?

File: utils/general.ts [hunk 6]
```diff
@@ -31,13 +24,27 @@ export const addOffset = (dates: Date[], { offsetStart, offsetEnd }: { offsetSta
...
+    const params = slugsFromUrl(pathname).params;
+
+    return params;
   } catch (err) {
-    console.error('paramsFromUrl error:', err);
+    console.error('error on paramsFromUrl():', err);
+    return;
   }
 };
+//#endregion
```
Comment (SgtPooki) [inline utils/general.ts:47]: ditto: whats this for?

File: pages/api/roadmap.ts [hunk 7]
```diff
@@ -89,33 +97,53 @@ const addToChildren = (data, parent) => {
...
+    if (!rootIssue) {
+      res.status(500).json({ error: { code: '500', message: 'No issue found.' } });
+      return;
+    }
+
+    // We should probably check for due dates of milestones instead of the root roadmap
     // if (!getConfig(rootIssue?.body_html)?.eta) {
     //   // res.status(500).json({ error: { code: '500', message: 'No due date found in issue body.' } });
     //   // return;
     // }
```
Comment (SgtPooki) [inline pages/api/roadmap.ts:128]: remove?

File: pages/api/roadmap.ts [hunk 8]
```diff
@@ -89,33 +97,53 @@ const addToChildren = (data, parent) => {
...
 
-    const childrenFromBodyHtml = (!!rootIssue && rootIssue.body_html && getChildren(rootIssue.body_html)) || null;
+    const childrenFromBodyHtml = rootIssue.body_html && getChildren(rootIssue.body_html);
+    // console.log('rootIssue:', rootIssue);
+    // console.log('childrenFromBodyHtml:', childrenFromBodyHtml);
```
Comment (SgtPooki) [inline pages/api/roadmap.ts:132]: remove

**Review by SgtPooki — CHANGES_REQUESTED** [review]

File: pages/roadmap-test/[...slug].tsx (hunk 2 above)
Comment (AlexxNica) [reply pages/roadmap-test/[...slug].tsx:1]: I didn't get that. Were you planning on using the `pages/roadmap-test`? That's an endpoint I created to do some testing but wasn't using for anything else. Note that this is not the standard `pages/roadmap`.

Comment (AlexxNica) [reply pages/roadmap-test/[...slug].tsx:1]: @SgtPooki I didn't get that. Were you planning on using the pages/roadmap-test? That's an endpoint I created to do some testing but wasn't using for anything else. Note that this is not the standard pages/roadmap.

File: pages/roadmap-test/index.tsx (hunk 3 above)
Comment (AlexxNica) [reply pages/roadmap-test/index.tsx:1]: Same as https://github.com/pln-planning-tools/Starmaps/pull/81#discussion_r1020592066

File: pages/roadmap-test/[...slug].tsx (hunk 2 above)
Comment (SgtPooki) [reply pages/roadmap-test/[...slug].tsx:1]: I also created those pages, maybe I didn't push my changes up? if those are yours, I'm fine with removing them

**Review by whizzzkid — CHANGES_REQUESTED** [review]

**Comment by SgtPooki** [comment]
FYI that we're putting this on pause so we can focus on P0 issues on https://github.com/orgs/pln-planning-tools/projects/2

**Comment by SgtPooki** [comment]
There are a lot of conflicts now. We will close this and address any concerns with smaller more isolated fixes. Thanks for trying to get this in before us stomping all over the code though @AlexxNica!

**Pull request closed without merging**